src/main/java/dog/pawbook/model/managedentity/dog/Dog.java

```diff
@@ -0,0 +1,125 @@
+package dog.pawbook.model.managedentity.dog;
+
+import static dog.pawbook.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+import dog.pawbook.model.managedentity.Entity;
+import dog.pawbook.model.managedentity.Name;
+import dog.pawbook.model.managedentity.owner.Owner;
+import dog.pawbook.model.managedentity.tag.Tag;
+
+
+/**
+ * Represents a dog in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Dog extends Entity {
+
+    enum Sex {
+        MALE,
+        FEMALE
+    }
+
+    //Identity Fields
+    private final Breed breed;
+    private final DateOfBirth dob;
+    private final Sex sex;
+
+    // Data fields
+    private final Owner owner;
```

Comment:
I think we came to a conclusion last week that this field is to be storing an ID of the owner.

Reply:
Alright i'll edit this

Reply:
Yup, the idea here is to take in the ID and verify with the backend that the owner does exist for this ID to be considered valid. But you can just put some placeholder functions now when it comes to the adding part.